### PR TITLE
New package: py-deeplabcut

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-deeplabcut/package.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    depends_on("py-filterpy", type=('build', 'run'))
+    depends_on("py-h5py", type=('build', 'run'))
+    depends_on("py-intel-openmp", type=('build', 'run'))
+    depends_on("py-imgaug", type=('build', 'run'))
+    depends_on("py-ipython", type=('build', 'run'))
+    depends_on("py-ipython-genutils", type=('build', 'run'))
+    depends_on("py-numba@0.51.1", type=('build', 'run'))
+    depends_on("py-matplotlib@3.1.3", type=('build', 'run'))
+    depends_on("py-moviepy@:1.0.1", type=('build', 'run'))
+    depends_on("py-numpy@1.16.4:1.999", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-deeplabcut/package.py]:
```suggestion
    depends_on("py-numpy@1.16.4", type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-deeplabcut/package.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    # depends_on("py-opencv-python-headless", type=('build', 'run'))
+    depends_on("opencv+python", type=('build', 'run'))
+    depends_on("py-pandas@1.0.1:", type=('build', 'run'))
+    depends_on("py-patsy", type=('build', 'run'))
+    depends_on("py-python-dateutil", type=('build', 'run'))
+    depends_on("py-pyyaml", type=('build', 'run'))
+    depends_on("py-requests", type=('build', 'run'))
+    depends_on("py-ruamel-yaml@0.15.0:", type=('build', 'run'))
+    depends_on("py-setuptools", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-deeplabcut/package.py]: You can remove this above so it's not duplicated

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by alalazo** [comment]
@qwertos Can you please rebase and ping me and @adamjstewart back?

**Comment by qwertos** [comment]
@alalazo @adamjstewart rebased and pinged

File: var/spack/repos/builtin/packages/py-deeplabcut/package.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    depends_on("py-intel-openmp", type=('build', 'run'))
+    depends_on("py-imgaug", type=('build', 'run'))
+    depends_on("py-ipython", type=('build', 'run'))
+    depends_on("py-ipython-genutils", type=('build', 'run'))
+    depends_on("py-numba@0.51.1", type=('build', 'run'))
+    depends_on("py-matplotlib@3.1.3", type=('build', 'run'))
+    depends_on("py-moviepy@:1.0.1", type=('build', 'run'))
+    depends_on("py-numpy@1.16.4", type=('build', 'run'))
+    # depends_on("py-opencv-python-headless", type=('build', 'run'))
+    depends_on("opencv+python", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-deeplabcut/package.py:39]: I changed the `opencv` package so now there is no `+python` variant. As far as I can tell, `deeplabcut` only supports Python 3? So I would change this:
```suggestion
    depends_on("opencv+python3", type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-deeplabcut/package.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    depends_on("py-pandas@1.0.1:1.1.999", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-deeplabcut/package.py:40]: Where does this upper bound come from?

File: var/spack/repos/builtin/packages/py-deeplabcut/package.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    networks that achieves excellent results (i.e. you can
+    match human labeling accuracy) with minimal training data
+    (typically 50-200 frames)."""
+
+    homepage = "http://www.mousemotorlab.org/deeplabcut"
+    pypi = "deeplabcut/deeplabcut-2.2b8.tar.gz"
+
+    version('2.1.10', sha256='457ae694ec91d4903d7077c4ef14a00444a9f8932f3d1cbe08d1c2d58c326df5')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-deeplabcut/package.py:21]: Can you check if this is needed at run-time?

File: var/spack/repos/builtin/packages/py-deeplabcut/package.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    depends_on("py-bayesian-optimization", type=('build', 'run'))
+    depends_on("py-certifi", type=('build', 'run'))
+    depends_on("py-chardet", type=('build', 'run'))
+    depends_on("py-click", type=('build', 'run'))
+    depends_on("py-cython", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-deeplabcut/package.py:26]: Can you check if this is needed at run-time?

**Comment by alalazo** [comment]
@qwertos

**Comment by alalazo** [comment]
Closing this PR as stale, but feel free to reopen if you want to continue working on it.

**Pull request closed without merging**